pkg/controller/migmigration/stage.go

```diff
@@ -543,7 +543,7 @@ func (t *Task) ensureStagePodsTerminated() (bool, error) {
 		corev1.PodFailed:    true,
 		corev1.PodUnknown:   true,
 	}
-	options := k8sclient.MatchingLabels(t.PlanResources.MigPlan.GetCorrelationLabels())
+	options := k8sclient.MatchingLabels(t.stagePodIdentifierLabel())
```

Comment:
We probably need namespace filtering here too.

Reply:
@sseago I addressed your comments in the latest change.